test_suite/tests/derive.rs

```diff
@@ -172,20 +190,39 @@ fn c_like_enum_derive_with_scale_index_set() {
 fn enum_derive() {
     #[allow(unused)]
     #[derive(TypeInfo)]
+    /// Enum docs.
     enum E<T> {
-        A(T),
-        B { b: T },
+        /// Unnamed fields variant.
+        A(
+            /// Unnamed field.
+            T,
+        ),
+        /// Named fields variant.
+        B {
+            /// Named field.
+            b: T,
+        },
+        /// Unit variant.
         C,
     }
```

Comment:
Would be good to test for things like:

- `#[doc(hidden)`
- `#[doc = " This is a doc comment."]`
- That we handle inner/outer docs properly, e.g. a misplaced `//! Wut?`
- `#[doc(primitive)]` or any other less common variants of the `#[doc]` attribute?